docs/BUILDING.md

```diff
@@ -0,0 +1,119 @@
+Building snmalloc
+=================
+
+snmalloc uses a CMake build system and can be built on many platforms.
+
+# Building on Windows
+
+The Windows build currently depends on Visual Studio 2017.
+To build with Visual Studio:
+
+```
+mkdir build
+cd build
+cmake -G "Visual Studio 15 2017 Win64" ..
+cmake --build . --config Debug
+cmake --build . --config Release
+cmake --build . --config RelWithDebInfo
+```
+
+You can also omit the last three steps and build from the IDE.
+Visual Studio builds use a separate directory to keep the binaries for each
+build configuration.
+
+Alternatively, you can follow the steps in the next section to build with Ninja
+using the Visual Studio compiler.
+
+# Building on UNIX-like platforms
+
+snmalloc has platform abstraction layers for XNU (macOS, iOS, and so on),
+FreeBSD, NetBSD, OpenBSD, and Linux and is expected to work out of the box on
+these systems.
+Please open issues if it does not.
+Note that NetBSD, by default, ships with a toolchain that emits calls to
+`libatomic` but does not ship `libatomic`.
+To use snmalloc on NetBSD, you must either acquire a `libatomic` implementation
+(for example, from the GCC or LLVM project) or compile with clang.
+
+snmalloc has very few dependencies, CMake, Ninja, Clang 6.0 or later and a C++17
+standard library.
+Building with GCC is currently not recommended because GCC emits calls to
+libatomic for 128-bit atomic operations.
+
+To build a debug configuration:
+```
+mkdir build
+cd build
+cmake -G Ninja .. -DCMAKE_BUILD_TYPE=Debug
+ninja
+```
+To build a release configuration:
```

Comment:
```suggestion
To build a release configuration (that is, one with optimizations turned on):
```

Reply:
Release also implies no asserts and no debug info, so I think this is a misleading change.

Reply:
Are asserts implied by `RelWithDebugInfo`?  Clarifying both branches might be worthwhile.

Reply:
No, asserts are based on the NDEBUG macro not being defined.  All of this is inherited from CMake and I'm hesitant to try to redefine core CMake configs in our build.

Reply:
Sorry, to be clear: I wasn't suggesting changing anything about CMake or snmalloc's use of it, I was just reacting to the next stanza in the docs, which describe `RelWithDebugInfo` as "with optimizations on but with debug info".  That doesn't make it clear whether it's Release-like in having shed assertions or Debug-like in having kept them.